projects/observability/src/pages/apis/service-detail/instrumentation/service-instrumentation.component.ts

```diff
@@ -26,15 +27,21 @@ import { ServiceScoreResponse } from './service-instrumentation.types';
 export class ServiceInstrumentationComponent implements OnInit {
   public constructor(
     private readonly breadcrumbsService: BreadcrumbsService,
-    private readonly serviceInstrumentationService: ServiceInstrumentationService
+    private readonly serviceInstrumentationService: ServiceInstrumentationService,
+    private readonly subscriptionLifecycle: SubscriptionLifecycle
   ) {}
 
   public ngOnInit(): void {
-    this.breadcrumbsService.getLastBreadCrumbString().subscribe(serviceName => {
-      this.serviceInstrumentationService
-        .getServiceScore(serviceName)
-        .subscribe(serviceScore => this.serviceInstrumentationService.serviceScoreSubject.next(serviceScore));
-    });
+    this.subscriptionLifecycle.add(
+      this.breadcrumbsService.getLastBreadCrumbString().subscribe(serviceName => {
+        serviceName !== 'Services' &&
```

Comment:
This doesn't look ideal, can't we fix this at the `getLastBreadCrumbString` itself?
Also, not sure why `Services` is returned here?

Reply:
Looks like this is happening because `breadcrumbs$` inside BreadcrumbsService is a `ReplayObservable`, so it will always emit all breadcrumb values no matter at what point you subscribe to it.

We need something that emits latest values, not all values. Is there a way to convert a ReplayObservable to a regular Observable or something else inside `getLastBreadCrumbString`?

Reply:
<img width="729" alt="image" src="https://user-images.githubusercontent.com/56739130/198337550-26121054-17ec-4193-836b-61c2ecc84969.png">

Unable to reproduce what you're saying on local. This method is always returning the respective `serviceName`.
Can you reproduce once and tell me the steps?

Reply:
Add this in the constructor of `service-instrumentation.component.ts`

```ts
this.serviceInstrumentationService.serviceScoreSubject.subscribe(v => console.log(v));
```

Reply:
Changed the hook to `AfterViewInit` and now I'm not getting the 'Services' value anymore. Maybe it just needed to wait for the breadcrumbs to finish loading/rendering before getting any values from BreadcrumbsService.

Reply:
Tested on stage. Seems to be working fine. Merging changes.